python/ffcc/io.py

```diff
@@ -14,145 +14,153 @@
+      'scene_name_weight.txt' does not exist, it will be set to 1.
+    rgb: RGB image (float32)
+    extended_feature: value of the extended feature for this sample (
+      float32). If not 'scene_name_extended_feature.txt' does not exist,
+      it will be set to 1.
   """
-  # pyformat: enable
-
+  # load the weight if the weight file exists
+  if tf.io.gfile.exists(os.path.join(dir_path, scene_name + '_weight.txt')):
+    weight = np.loadtxt(os.path.join(dir_path, scene_name + '_weight.txt'),
+                        dtype=np.float32)
+  else:
+    weight = np.float32(1.0)
+  # load the extended feature if the weight file exists
+  if tf.io.gfile.exists(os.path.join(dir_path,
+                                     scene_name + '_extended_feature.txt')):
+    extended_feature = np.loadtxt(os.path.join(
+      dir_path, scene_name + '_extended_feature.txt'), dtype=np.float32)
+  else:
+    extended_feature = np.float32(1.0)
+  # load image and illuminant data
+  rgb = tf.io.decode_image(tf.io.read_file(
+    os.path.join(dir_path, scene_name + '.png')), dtype=tf.dtypes.uint16)
+  normalized = tf.cast(rgb, dtype=tf.float32) / tf.uint16.max
+  tf.compat.v1.disable_eager_execution()
+  rgb = normalized.eval(session=tf.compat.v1.Session())
   # Make sure the illuminant is unit vector.
   rgb_illuminant = np.loadtxt(
-      os.path.join(dir_path, 'illuminant.txt'), dtype=np.float32)
+    os.path.join(dir_path, scene_name + '.txt'), dtype=np.float32)
   rgb_illuminant /= np.linalg.norm(rgb_illuminant)
-  weight = np.loadtxt(os.path.join(dir_path, 'weight.txt'), dtype=np.float32)
-
-  frames = []
-  for item in tf.io.gfile.listdir(dir_path):
-    sub_dir = os.path.join(dir_path, item)
-    if tf.io.gfile.isdir(sub_dir) and item.isdigit():
-      try:
-        rgb = sio.imread(os.path.join(sub_dir, 'image.tiff'))
-        extended_feature = np.loadtxt(
-            os.path.join(sub_dir, 'extended_feature.txt'), dtype=np.float32)
-        frames.append((rgb, extended_feature))
-      except:  # pylint: disable=bare-except
-        tf.logging.error('Unable to load %s', sub_dir)
+  return rgb_illuminant, weight, rgb, extended_feature
 
-  return rgb_illuminant, weight, frames
 
-
-def encode_to_tfrecord(data, output_dir):
-  """Encodes a data item to TFRecord format.
+def get_training_eval_sets(files, cvfolds, test_fold):
+  """Retrieve train and evaluation sets for the current testing fold .
 
   Args:
-    data: A dictionary maps to the FFCC data bundle.
-    output_dir: The output folder. The record will be saved as <name>.tfrecord.
+    files: A list of filenames that should be synched with the cvfolds.
+    cvfolds: A list of corresponding fold number of each filename in 'files'
+    test_fold: Number of testing fold (should be in the range [1-3]).
+
+  Returns:
+    train_set and eval_set, lists of training and evaluation
+      filenames, respectively.
   """
+  assert len(files) == len(cvfolds)
+  assert test_fold in range(1,4)
```

Comment:
I think this can easily be made more generic by passing in the number of expected folds to this function and making it a project parameter. Or just to keep it simple, hardcode the number of folds and check as:

assert test_fold in range(1, NUM_CROSS_VALIDATION_FOLDS + 1)